airgun/entities/host_new.py

```diff
@@ -23,6 +26,81 @@ def get_details(self, entity_name, widget_names=None):
         view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
         return view.read(widget_names=widget_names)
 
+    def get_packages(self, entity_name, search=""):
+        """Filter installed packages on host"""
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.Content.Packages.select()
+        view.Content.Packages.searchbar.fill(search)
+        # wait for filter to apply
+        time.sleep(1)
+        return view.Content.Packages.read()
+
+    def install_package(self, entity_name, package):
+        """Installs package on host using the installation modal"""
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.Content.Packages.select()
+        view.Content.Packages.dropdown.wait_displayed()
+        view.Content.Packages.dropdown.item_select('Install packages')
+        view = InstallPackagesView(self.browser)
+        view.wait_displayed()
+        view.searchbar.fill(package)
+        # wait for filter to apply
+        time.sleep(2)
```

Comment:
I'm not against/harsh the hard sleep in a single place, but that is the consistent multiple places we really need to think about what causes such behavior/pattern. Can you use the before/after_fill to handle such cases (use the wait_for)?

Reply:
So the problem here is that basically after filling the filter the table is still visible with old data. So I am not sure for what should I wait, as the table is visible and everything seems to be "correct". Maybe I can use after_fill but still, it will need to contain the sleep. I have tried avoiding using sleep, but haven't figured out a better way, if you have some ideas let me know.

Reply:
The `ensure_page_safe` checks should be handling this automatically in the browser plugin.

Something about the page should trigger the checks in this function, letting us know that the page isn't ready to interact with yet:
https://github.com/SatelliteQE/airgun/blob/986ff09d6a578414d5f4099cd776082a9f1d77a9/airgun/browser.py#L192